include/api_info.hpp

```diff
@@ -102,10 +167,14 @@ struct ApiInfo {
 			return 24;
 		case SQL_TYPE_DATE:
 			return 10;
-		case SQL_TYPE_TIME:
-			return 9;
-		case SQL_TYPE_TIMESTAMP:
-			return 20;
+		case SQL_TYPE_TIME: {
+			uint8_t precision = GetTemporalPrecision(logical_type.id());
+			return precision > 0 ? 9 + precision : 8; // HH:MM:SS[.fff...]
+		}
+		case SQL_TYPE_TIMESTAMP: {
+			uint8_t precision = GetTemporalPrecision(logical_type.id());
+			return precision > 0 ? 20 + precision : 19; // YYYY-MM-DD HH:MM:SS[.fff...]
```

Comment:
If I am not mistaken, the logic for timestamp and time in `GetColumnSize` and in `GetDisplaySize` is exactly the same, just presented differently. Can we use the same calculations (I think helpers are not really necessary) in both functions?

Reply:
I'm not following @staticlibs - could you clarify what specific changes you're looking for? I'd be happy to refactor this once I understand what you have in mind. Thanks

Reply:
Ah yes, good eye! I had change blindness on that one. I went with the style of the first in my latest commit as it feels simpler to follow